bin dist type with no output panics

I'm trying to run a manual `dist` task without a `build` task for a product and it panics:
config:
```
products:
  project:
    dist:
      disters:
        type: manual
        config:
          extension: tgz
        script: |
          #!/usr/bin/env bash
          echo "hi"
```
generates this:
```
➜  project git:(gd/initial2) ✗ ./godelw dist planer
Creating distribution for planer at out/dist/planer/unspecified/bin/project-unspecified.tgz
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x13f9aaa]

goroutine 1 [running]:
github.com/palantir/distgo/dister/bin.(*Dister).RunDist(0x1975888, 0xc42002b607, 0x3, 0x7ffeefbff4fd, 0x3d, 0x160719f, 0xb, 0xc42002b640, 0x6, 0x0, ...)
	/go/src/github.com/palantir/distgo/dister/bin/dister.go:52 +0x3a
github.com/palantir/distgo/distgo/dist.Run(0x7ffeefbff4fd, 0x3d, 0x160719f, 0xb, 0xc42002b640, 0x6, 0x0, 0x0, 0xc420182b20, 0x0, ...)
	/go/src/github.com/palantir/distgo/distgo/dist/dist.go:160 +0x73b
github.com/palantir/distgo/distgo/dist.Products(0x7ffeefbff4fd, 0x3d, 0x160719f, 0xb, 0xc42015d420, 0x0, 0x0, 0x1672fe0, 0x1975888, 0x1671940, ...)
	/go/src/github.com/palantir/distgo/distgo/dist/dist.go:107 +0xc6d
github.com/palantir/distgo/cmd.glob..func6(0x1950700, 0xc42013f340, 0x1, 0x7, 0x0, 0x0)
	/go/src/github.com/palantir/distgo/cmd/dist.go:41 +0x250
github.com/palantir/distgo/vendor/github.com/spf13/cobra.(*Command).execute(0x1950700, 0xc42012e240, 0x7, 0xc, 0x1950700, 0xc42012e240)
	/go/src/github.com/palantir/distgo/vendor/github.com/spf13/cobra/command.go:756 +0x468
github.com/palantir/distgo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1951a00, 0xc42008df80, 0x1604def, 0x5)
	/go/src/github.com/palantir/distgo/vendor/github.com/spf13/cobra/command.go:846 +0x30a
github.com/palantir/distgo/vendor/github.com/palantir/pkg/cobracli.Execute(0x1951a00, 0xc42008df50, 0x3, 0x3, 0x0)
	/go/src/github.com/palantir/distgo/vendor/github.com/palantir/pkg/cobracli/execute.go:30 +0xea
github.com/palantir/distgo/vendor/github.com/palantir/pkg/cobracli.ExecuteWithDebugVarAndDefaultParams(0x1951a00, 0x1975630, 0x0, 0x0, 0x0, 0xc420059f78)
	/go/src/github.com/palantir/distgo/vendor/github.com/palantir/pkg/cobracli/default.go:36 +0x10d
github.com/palantir/distgo/cmd.Execute(0xc42001c0a0)
	/go/src/github.com/palantir/distgo/cmd/root.go:60 +0x4a
main.main()
	/go/src/github.com/palantir/distgo/main.go:34 +0xe6
```

--- a/dister/bin/dister.go
+++ b/dister/bin/dister.go
@@ -49,6 +49,10 @@ func (d *Dister) PackagingExtension() (string, error) {
 }
 
 func (d *Dister) RunDist(distID distgo.DistID, productTaskOutputInfo distgo.ProductTaskOutputInfo) ([]byte, error) {
+	if productTaskOutputInfo.Product.BuildOutputInfo == nil {
+		return nil, errors.Errorf("bin dist failed: no build outputs for product %s", productTaskOutputInfo.Product.ID)
+	}
+
 	for _, osArch := range productTaskOutputInfo.Product.BuildOutputInfo.OSArchs {
 		if err := verifyDistTargetSupported(osArch, productTaskOutputInfo); err != nil {
 			return nil, err

--- a/dister/bin/integration_test/integration_test.go
+++ b/dister/bin/integration_test/integration_test.go
@@ -306,6 +306,25 @@ Finished creating bin distribution for foo
 					verifyLayoutAndTGZ(t, projectDir, wantInnerLayout)
 				},
 			},
+			{
+				Name: "bin does not panic if build output does not exist",
+				ConfigFiles: map[string]string{
+					"godel/config/godel.yml": godelYML,
+					"godel/config/dist-plugin.yml": `
+products:
+  foo:
+    dist:
+      disters:
+        type: bin
+`,
+				},
+				WantError: true,
+				WantOutput: func(projectDir string) string {
+					return `Creating distribution for foo at out/dist/foo/1.0.0/bin/foo-1.0.0.tgz
+Error: dist failed for foo: bin dist failed: no build outputs for product foo
+`
+				},
+			},
 		},
 	)
 }

